Core/src/Components/AudioComponent.cpp

```diff
@@ -1,78 +1,73 @@
-}
+AudioComponent::AudioComponent(Entity* mEntity, AudioManager* audioManager, FMOD::ChannelGroup* channelGroup) : Component(mEntity) {
+	AudioComponent::previousPosition = entity->transform->getWorldPosition();
+	AudioComponent::am = audioManager;
+	AudioComponent::group = channelGroup;
+	this->componentType = ComponentType::Audio;
+};
 
-void AudioComponent::PlayOnAwake(bool play)
-{
-    playOnAwake = play;
-}
+AudioComponent::AudioComponent(Entity* mEntity, AudioManager* audioManager, FMOD::Sound* sound) : AudioComponent(mEntity, audioManager) {
+	FMOD_VECTOR pos = AudioManager::GetFMODVector(previousPosition);
+	// FMOD_VECTOR forward = AudioManager::GetFMODVector(entity->transform->getForward());
+	// FMOD_VECTOR up = AudioManager::GetFMODVector(entity->transform->getUp());
+	am->GetSystem()->playSound(sound, group, true, &channel);
+	channel->set3DAttributes(&pos, nullptr);
+};
 
-void PlayOnAwake(bool playOnAwake);
 
-void AudioComponent::LoadAudio() {
-    if (!audioManager) {
-        std::cerr << "AudioManager not available!" << std::endl;
-        return;
-    }
+AudioComponent::AudioComponent(Entity* mEntity, AudioManager* audioManager, FMOD::ChannelGroup* channelGroup, FMOD::Sound* sound) : AudioComponent(mEntity, audioManager, channelGroup) {
+	FMOD_VECTOR pos = AudioManager::GetFMODVector(previousPosition);
+	FMOD_VECTOR forward = AudioManager::GetFMODVector(entity->transform->getForward());
+	FMOD_VECTOR up = AudioManager::GetFMODVector(entity->transform->getUp());
+	FMOD_RESULT result = am->GetSystem()->playSound(sound, group, true, &channel);
+	if (result == FMOD_OK)
+		channel->setChannelGroup(group);
+	channel->set3DAttributes(&pos, nullptr);
+};
 
-    sound = audioManager->LoadAudio(filePath);
-    if (!sound) {
-        std::cerr << "Failed to load audio: " << filePath << std::endl;
-    }
-}
+AudioComponent::~AudioComponent() {
 
-void AudioComponent::Play() {
-    if (!sound) {
-        LoadAudio();
-    }
+};
 
-    if (sound) {
-        audioManager->PlaySound(sound, isLooping, channel, entity->transform->getLocalPosition());
-        playOnAwake = true;
-    }
-}
+void AudioComponent::PlaySound(FMOD::Sound* sound, bool isLooping, bool isPaused) {
+	if (!sound) return;
+	if (isLooping)
+		sound->setMode(FMOD_LOOP_NORMAL);
+	FMOD_RESULT result = am->GetSystem()->playSound(sound, group, true, &channel);
+	if (result == FMOD_OK && channel) {
+		channel->setLoopCount(-1);
+		FMOD_VECTOR pos = AudioManager::GetFMODVector(entity->transform->getWorldPosition());
+		channel->set3DAttributes(&pos, nullptr);
+		channel->setPaused(isPaused);
+	}
+};
 
-void AudioComponent::Stop() {
-    if (channel) {
-        channel->stop();
-        channel = nullptr;
-        playOnAwake = false;
-    }
-}
+void AudioComponent::UpdateAttributes(float deltaTime) {
+	glm::vec3 currentPosition = entity->transform->getWorldPosition();
+	FMOD_VECTOR vel = AudioManager::GetFMODVector((currentPosition - previousPosition) / deltaTime);
```

Comment:
We never set previous position to a new value.

Reply:
Good catch, thanks. Will do.